src/main/java/sky/command/DeadlineCommand.java

```diff
@@ -0,0 +1,107 @@
+package sky.command;
+import java.io.IOException;
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.regex.PatternSyntaxException;
+
+import sky.TaskList;
+import sky.exception.TextNoMeaningException;
+import sky.storage.History;
+import sky.storage.Storage;
+import sky.task.Deadline;
+import sky.task.Task;
+
+/**
+ * The DeadlineCommand class deals with adding a deadline task into taskList.
+ */
+public class DeadlineCommand extends Command {
+    private String fullCommand;
+
+    public DeadlineCommand(String fullCommand) {
+        this.fullCommand = fullCommand;
+    }
+
+    @Override
+    public String execute(TaskList taskList, Storage storage, History history)
+            throws TextNoMeaningException, IOException {
+        try {
+            String[] arrOfStrings = generateTaskDescriptionAndUserInputBy();
+            assert arrOfStrings.length > 0 : "arrOfStrings should not be empty.";
+            String taskDescription = arrOfStrings[0];
+            String taskByUserInput = arrOfStrings[1];
+            String taskBy = produceDateAndTimeForDeadline(taskByUserInput);
+            Task task = new Deadline(taskDescription, taskBy);
+            taskList.addTask(task);
+            // Add task into data file.
+            storage.append(task.toString());
+            history.addHistoryInTime(taskList);
+            return generateResponse(task, taskList);
+        } catch (IndexOutOfBoundsException e) {
+            throw new TextNoMeaningException("You have either not entered any text after typing deadline, "
+                    + "or you have positioned your slash wrongly.");
+        } catch (PatternSyntaxException e) {
+            throw new TextNoMeaningException("There is a problem with the regex expression written by the dev.");
+        }
+    }
+
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+
+    private String produceDateAndTimeForDeadline(String s) throws TextNoMeaningException {
+        try {
+            String[] arrOfStrings = s.split(" ");
+            if (arrOfStrings.length != 1 && arrOfStrings.length != 2) {
+                throw new TextNoMeaningException("Provide the date and time after \"/by\""
+                        + " as: \"yyyy/mm/dd XXXX\", where XXXX is time in 24-hours.");
+            }
+            String dateGiven = arrOfStrings[0].replaceAll("/", "-");
+            LocalDate d1 = LocalDate.parse(dateGiven);
+            String dateString = d1.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+
+            // If time is provided
+            if (arrOfStrings.length == 2) {
```

Comment:
I noticed that there are comments in your code that explains **how** the code works. Perhaps you can consider ways to abstract these out into another method to apply SLAP or avoid magic literals by abstracting out these variables? 
```suggestion
            boolean isTimeProvided = arrOfStrings.length == 2
```

Reply:
Thanks for the suggestion! It's a good one and I am working on it now 😄